File: app/src/main/java/ir/ghararemaghzha/game/fragments/InviteFragment.java
package ir.ghararemaghzha.game.fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

import com.akexorcist.roundcornerprogressbar.RoundCornerProgressBar;
import com.google.android.material.textview.MaterialTextView;

import ir.ghararemaghzha.game.R;
import ir.ghararemaghzha.game.classes.MySharedPreference;
import ir.ghararemaghzha.game.classes.Utils;
import ir.ghararemaghzha.game.data.RetrofitClient;
import ir.ghararemaghzha.game.models.GeneralResponse;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class InviteFragment extends Fragment {
    private Context context;
    private FragmentActivity activity;
    private MaterialTextView count, value;
    private RoundCornerProgressBar progressBar;
    private float fCount = 0f;
    private ConstraintLayout loading;

    public InviteFragment() {
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_invite, container, false);
        context = getContext();
        activity = getActivity();

        init(v);
        return v;
    }

    private void init(View v) {
        ((MaterialTextView) activity.findViewById(R.id.toolbar_title)).setText(R.string.invite_title);
        loading = v.findViewById(R.id.invite_loading);
        count = v.findViewById(R.id.invite_count);
        value = v.findViewById(R.id.invite_info_text1);
        progressBar = v.findViewById(R.id.invite_progress);

        ((MaterialTextView) v.findViewById(R.id.invite_code)).setText(MySharedPreference.getInstance(context).getUserCode());
        getData();
        onClicks(v);
    }

    private void onClicks(View v) {
        v.findViewById(R.id.invite_share).setOnClickListener(vv -> Utils.shareCode(activity, getString(R.string.invite_share, MySharedPreference.getInstance(context).getUserCode())));
    }

    private void getData() {
        String number = MySharedPreference.getInstance(context).getNumber();
        String token = MySharedPreference.getInstance(context).getAccessToken();
        if (number.isEmpty() || token.isEmpty()) {
            Utils.logout(activity, true);
            return;
        }
        loading.setVisibility(View.VISIBLE);

        RetrofitClient.getInstance().getApi()
                .getInvites("Bearer " + token, number)
                .enqueue(new Callback<GeneralResponse>() {
                    @Override
                    public void onResponse(@NonNull Call<GeneralResponse> call, @NonNull Response<GeneralResponse> response) {
                        loading.setVisibility(View.GONE);

                        if (response.isSuccessful() && response.body() != null && response.body().getResult().equals("success")) {
                            fCount = Float.parseFloat(response.body().getCount())+0f;
                            int remaining = 10 - (int) fCount;
                            count.setText(getString(R.string.invite_star_title, remaining));
                            progressBar.setProgress(fCount);
                            value.setText(getString(R.string.invite_info_text1, response.body().getValue()));

                        } else if (response.code() == 401) {
                            Utils.logout(activity, true);
                        } else {
                            Utils.showInternetError(context, () -> getData());
                        }
                    }

                    @Override
                    public void onFailure(@NonNull Call<GeneralResponse> call, @NonNull Throwable t) {
                        loading.setVisibility(View.GONE);
                        Utils.showInternetError(context, () -> getData());

                    }
                });


    }
}

File: app/src/main/java/ir/ghararemaghzha/game/fragments/InviteFragment.kt
package ir.ghararemaghzha.game.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.navigation.NavController
import androidx.navigation.Navigation
import com.google.android.material.textview.MaterialTextView
import ir.ghararemaghzha.game.R
import ir.ghararemaghzha.game.classes.MySharedPreference
import ir.ghararemaghzha.game.classes.Utils
import ir.ghararemaghzha.game.classes.Utils.Companion.shareCode
import ir.ghararemaghzha.game.classes.Utils.Companion.showInternetError
import ir.ghararemaghzha.game.data.Resource
import ir.ghararemaghzha.game.databinding.FragmentInviteBinding
import ir.ghararemaghzha.game.viewmodels.InviteViewModel

class InviteFragment : BaseFragment<InviteViewModel, FragmentInviteBinding>() {
    private lateinit var navController: NavController
    private lateinit var number: String
    private lateinit var token: String


    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        init()

        viewModel.invitesResponse.observe(viewLifecycleOwner, { res ->
            when (res) {
                is Resource.Success -> {
                    if (res.value.result == "success") {
                        val fCount = res.value.count.toFloat()
                        val remaining = 20 - fCount.toInt()
                        b.inviteCount.text = getString(R.string.invite_star_title, remaining)
                        b.inviteProgress.progress = fCount
                        b.inviteInfoText1.text = getString(R.string.invite_info_text1, res.value.value)
                        b.inviteLoading.visibility = View.GONE

                    } else {
                        showInternetError(requireContext()) { viewModel.getInvites("Bearer $token", number) }
                        Toast.makeText(context, getString(R.string.general_error), Toast.LENGTH_SHORT).show()
                    }
                }
                is Resource.Failure -> {
                    if (res.isNetworkError) {
                        showInternetError(requireContext()) { viewModel.getInvites("Bearer $token", number) }
                        Toast.makeText(context, getString(R.string.general_error), Toast.LENGTH_SHORT).show()
                    } else if (res.errorCode == 401) {
                        b.inviteLoading.visibility = View.GONE
                        Utils.logout(requireActivity(), true)
                    }
                }
                is Resource.Loading -> {
                }

            }
        })
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        navController = Navigation.findNavController(view)
    }

    private fun init() {
        requireActivity().findViewById<MaterialTextView>(R.id.toolbar_title).setText(R.string.invite_title)
        b.inviteCode.text = MySharedPreference.getInstance(requireContext()).getUserCode()
        b.inviteShare.setOnClickListener {
            shareCode(requireContext(), getString(R.string.invite_share, MySharedPreference.getInstance(requireContext()).getUserCode()), "")
        }
        b.inviteContacts.setOnClickListener {
            navController.navigate(R.id.action_inviteFragment_to_contactsFragment)
        }
        b.inviteLoading.visibility = View.VISIBLE

        number = MySharedPreference.getInstance(requireContext()).getNumber()
        token = MySharedPreference.getInstance(requireContext()).getAccessToken()
        if (number.isEmpty() || token.isEmpty()) {
            Utils.logout(requireActivity(), true)
            return
        }

        viewModel.getInvites("Bearer $token", number)

    }

    override fun getViewModel() = InviteViewModel::class.java

    override fun getFragmentBinding(inflater: LayoutInflater, container: ViewGroup?) = FragmentInviteBinding.inflate(inflater, container, false)
}